### PR TITLE
refactor access logs to use separate functions listener access logs

### PR DESCRIPTION
**Pull request by ramaraochavali**
Similar to how `buildListenerFileAccessLog` - Implements the same for others for readability and consistency

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/accesslog.go [hunk 1]
```diff
@@ -94,19 +94,37 @@ func (b *AccessLogBuilder) setTCPAccessLog(push *model.PushContext, proxy *model
...
+			ConfigType: c.AccessLog.ConfigType,
+			Filter:     buildAccessLogFilter(filters...),
 		}
 
+		als = append(als, al)
+	}
+	return als
+}
+
+func buildListenerAccessLogFromTelemetry(cfgs []model.LoggingConfig) []*accesslog.AccessLog {
```
Comment (zirain) [inline pilot/pkg/networking/core/v1alpha3/accesslog.go:121]: this makes more duplicate?

**Comment by ramaraochavali** [comment]
@zirain PTAL when you get chance.

**Review by zirain — COMMENTED** [review]
this change seems more complex to me than before personally, I would like leave to @istio/wg-networking-maintainers

**Comment by ramaraochavali** [comment]
I tried to follow the same pattern we used for buildListenerFileAccessLog here also to be consistent and avoid code like if listener do some thing. But if folks think, this is more complex I can drop this PR

**Comment by ramaraochavali** [comment]
@howardjohn @hzxuzhonghu WDYT about this?

**Pull request closed without merging**